### PR TITLE
Change method of choosing next available config..

### PR DESCRIPTION
**Pull request by ingwarsw**
Made it much more randomised..

### CODE REVIEW

File: src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java [hunk 1]
```diff
@@ -271,6 +270,16 @@ public void addConfiguration(InstanceConfiguration configuration) {
     return result;
   }
 
+  /**
+   * Choose semi randomly config from list of available configs.
```
Comment (craigdbarber) [inline src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java]: This documentation seems misleading, as this is round robin not random.

Comment (ingwarsw) [reply src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java]: Its semi round robin.. cause it shares ID with all configs... so its random round robin :)

At current moment 99% of instances ends up always with first config..
Here all configs should be used rather uniformly..

I can make it full random.. 
But then there is huge problem with testing it..

Comment (ingwarsw) [reply src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java]: I can change javadoc if you want..

File: src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java [hunk 2]
```diff
@@ -271,6 +270,19 @@ public void addConfiguration(InstanceConfiguration configuration) {
...
+   * Choose config from list of available configs.
+   * Current implementation use round robin strategy starting at semi random element of list.
+   * Because most of times arriving requests asks for only 1 new node, we dont want 
+   * to start every time from 1 element.
+   *
+   * @param configs List of configs to choose from.
+   * @return Chosen config from list.
+   */
+  private InstanceConfiguration chooseConfigFromList(List<InstanceConfiguration> configs) {
+    return configs.get(Math.abs(configsNext++) % configs.size());
```
Comment (rachely3n) [inline src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java]: Isn't this the same method as before? Or are you going to configure the value of configsNext somewhere?

Comment (ingwarsw) [reply src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java]: This time its static value.. so it will change with every run..

Comment (rachely3n) [reply src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java]: Cool. Learned something new.

File: src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java (hunk 1 above)
Comment (craigdbarber) [reply src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java]: I think some additional clarification along those lines would be helpful. Thanks.

**Comment by rachely3n** [comment]
@ingwarsw Is there a reason for re-opening?

**Comment by ingwarsw** [comment]
Tried to force rebuilding..
But its not working :|

**Pull request merged**